# Profitable App Profiles for the App Store and Google Play Markets

This project is part of the [Dataquest.io](https://www.dataquest.io) guided project. The goal is to find the profitable app profiles on both most popular mobile platforms – App Store and Google Play Markets.

My goal is to put into practice all the knowledge I gained in the frist steps of my *Data Analyst in Python path.* The first couse of the path was focusing on:
- Introduction to Python
    - Variables & Data Types
    - Lists, Dictionaries, Tuples
    - Loops, Functions, Conditional Statemens
- Introduction to Jupyter Notebook

**The assignment** of the project is to help developers understand what type of apps are likely to attract more users on Google Play and the App Store.

## The datasets
This project use two publicly avaiable datasets:
- [Kaggle Dataset](https://www.kaggle.com/lava18/google-play-store-apps) containing data about approximately 10,000 Android apps from Google Play; the data was collected in August 2018. You can download the data set directly from this [link](https://dq-content.s3.amazonaws.com/350/googleplaystore.csv).
- [Kaggle Dataset](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps) containing data about approximately 7,000 iOS apps from the App Store; the data was collected in July 2017. You can download the data set directly from this [link](https://dq-content.s3.amazonaws.com/350/AppleStore.csv)


In [2]:
from csv import reader

# Opening the App store Dataset and creating list of lists
opened_file = open("AppleStore.csv")
read_file = reader(opened_file)
applestore = list(read_file)
apple_header = applestore[0] # list containing header row
apple = applestore[1:] #list containg data rows

# Opening the Google Play store and creating list of lists
opened_file = open("googleplaystore.csv")
read_file = reader(opened_file)
googleplay = list(read_file)
google_header = googleplay[0] # list containing header row
google = googleplay[1:] #list containg data rows

Function explore_data created to easily explore datasets.

In [3]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

### Exploring the datasets
**Apple Store**

In [4]:
print(apple_header)
print('\n')
explore_data(apple,0,5,True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


Number of rows: 7197
Number of columns: 16


The Apple Store datasets have 7197 apps with 16 columns. For the purpose of this analysis colums c

**Google Play**

In [5]:
print(google_header)
print('\n')
explore_data(google,0,5,True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Eve

On the [Kaggle discussion](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015) a mistake in the Google Play Store dataset is reported. Apparently one app is missing it's rating entry. To determine what is the index row in the google dataset and what app is missing the rating following code will give us the answer.

In [6]:
for row in google:
    if len(row) != len(google_header):
        print("Missing entry on row index: ",google.index(row))
        print("Lenght of the row: ",len(row))
        print("Lenght of the header: ", len(google_header))
        print(row)
        

Missing entry on row index:  10472
Lenght of the row:  12
Lenght of the header:  13
['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


The error row is on index number 10472, app name is *'Life Made WI-Fi Touchscreen Photo Frame'*. Let's delete this row to clean the data. Once the row is deleted check that we have correct dataset lenght. It should be 10840 since before it was 10841. The app name shoudn't match our finiding.

In [7]:
del google[10472]

In [8]:
for row in google:
    name = row[0]
    if name == 'Life Made WI-Fi Touchscreen Photo Frame':
        print("Error row was not deleted")

In [9]:
print("Actual Google dataset lenght: ",len(google))

Actual Google dataset lenght:  10840


## Deleting duplicates
As mentioned on the [Kaggle discussion](https://www.kaggle.com/lava18/google-play-store-apps/discussion/67894), some apps in the Google Play store dataset have duplicite entries. Let's check (randomly) if Facebook have duplicties, since it's one of the widely used apps.

In [10]:
for row in google:
    name = row[0]
    if name == "Facebook":
        print(row)
        print("Index row: ",google.index(row))

['Facebook', 'SOCIAL', '4.1', '78158306', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'August 3, 2018', 'Varies with device', 'Varies with device']
Index row:  2544
['Facebook', 'SOCIAL', '4.1', '78128208', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'August 3, 2018', 'Varies with device', 'Varies with device']
Index row:  3943


Finding apps randomly would consume lot of time. Instead I have created two lists. One contains unique apps and the second contains duplicates.

In [11]:
duplicates = []
unique_apps = []

for row in google:
    name = row[0]
    if name in unique_apps:
        duplicates.append(name)
    else:
        unique_apps.append(name)
        
print("Number of duplicate apps: ", len(duplicates))
print("Number of unique apps: ", len(unique_apps))
print("Control check: ", len(duplicates + unique_apps))
print('\n')
print("Examples of duplicates:", duplicates[:10])

Number of duplicate apps:  1181
Number of unique apps:  9659
Control check:  10840


Examples of duplicates: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack']


The data from the Facebook example show that the number of reviews differ. This will be used as a criterion to delete the duplicates. In the Google dataset we keep only the app with the highest number of reviews. 

Let's create a dictionary that will store unique apps with the highest number of reviews

In [12]:
reviews_max = {}

for row in google:
    name = row[0]
    n_reviews = float(row[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    if name not in reviews_max:
        reviews_max[name] = n_reviews


Inspecting the dictionary to make sure it contains all 9659 apps.

In [13]:
print("Dictionary lenght: ", len(reviews_max))

Dictionary lenght:  9659


The dictionary lenght matches the lenght of the unique apps. Let's proceed to deleteding the duplicates.

In [14]:
google_clean = [] #empty list for clean data
google_already_added = [] #empty list for the duplicates

for row in google:
    name = row[0]
    n_reviews = float(row[3])
    if (n_reviews == reviews_max[name]) and (name not in google_already_added):
        google_clean.append(row)
        google_already_added.append(name)

Inspecting the lists to see whether the lenght is matching our expectations. Google_clean should contain 9659 rows.

In [15]:
print("Lenght of clean dataset: ", len(google_clean))

Lenght of clean dataset:  9659


Checking for duplicates in the App Store dataset

In [16]:
duplicates_ios = []
unique_apps_ios = []

for row in apple:
    name = row[1]
    if name in unique_apps_ios:
        duplicates.append(name)
    else:
        unique_apps_ios.append(name)
        
print("Number of duplicate apps: ", len(duplicates_ios))
print("Number of unique apps: ", len(unique_apps_ios))
print("Control check: ", len(duplicates_ios + unique_apps_ios))

Number of duplicate apps:  0
Number of unique apps:  7195
Control check:  7195


There are no duplicities in the App Store dataset.

## Removing Non-English Apps
The imaginary company from the assigments focuses on developing apps in English. The goal is to analyze apps that are directed toward an English-speaking audience. Some of the app names are suggesting that they are not directed towards an English-speaking audience. See examples below:

In [17]:
print("Name: ", apple[813][1])
print("Name: ", apple[6731][1])
print('\n')
print("Name: ", google_clean[4412][0])
print("Name: ", google_clean[7940][0])

Name:  爱奇艺PPS -《欢乐颂2》电视剧热播
Name:  【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜


Name:  中国語 AQリスニング
Name:  لعبة تقدر تربح DZ


To clean both datasets we need to get rid of the non-English characters. The commonly used English text characters range 0 to 127 in [ASCII](https://en.wikipedia.org/wiki/ASCII). If the number in the app name is equal or less than 127, then the character belongs to the set of common English characters.

In [18]:
def is_english(string):
    non_ascii = 0
    
    for character in string:
        if ord(character) > 127:
            non_ascii += 1
            
    if non_ascii > 3:
        return False
    else:
        return True

print(is_english("Instagram"))
print(is_english("爱奇艺PPS -《欢乐颂2》电视剧热播"))
print(is_english("Docs To Go™ Free Office Suite"))
print(is_english("Instachat 😜"))

True
False
True
True


The app *Docs To Go™ Free Office Suite* is English but because of the ™ is part of the name the function evaluates it as False. To reduce data lost we will remove only apps that have more than 3 non ASCII characters.

In [19]:
apple_ascii = []
apple_non_ascii = []
google_ascii = []
google_non_ascii =[]

for row in apple:
    name = row[1]
    if is_english(name) == True:
        apple_ascii.append(row)
    else:
        apple_non_ascii.append(row)

for row in google_clean:
    name = row[0]
    if is_english(name) == True:
        google_ascii.append(row)
    else:
        google_non_ascii.append(row)

# Printing results and control
print("Number of English apps in App Store: ", len(apple_ascii))
print("Control count for App Store: ", len(apple_ascii) + len(apple_non_ascii))
print("Expected count for App Store: ", len(apple))
print("Number of English apps in Google Play: ", len(google_ascii))
print("Control count for Google Play: ", len(google_ascii) + len(google_non_ascii))
print("Expected count for Google Play: ", len(google_clean))

Number of English apps in App Store:  6183
Control count for App Store:  7197
Expected count for App Store:  7197
Number of English apps in Google Play:  9614
Control count for Google Play:  9659
Expected count for Google Play:  9659


There are 6183 English apps in the App Store and 9614 English apps in Google Play. The assignments later specifty that our developers are building only apps that are free to download and use. In the next step we will isolate only Free apps.

## Isolating free apps

In [20]:
apple_free = []
google_free = []

for row in apple_ascii:
    price = float(row[4])
    if price == 0.0:
        apple_free.append(row)
        
for row in google_ascii:
    price_type = row[6]
    if price_type == "Free":
        google_free.append(row)
        
print("Number of English, free apps in App Store: ", len(apple_free))
print("Number of English, free apps in Google Play: ", len(google_free))

Number of English, free apps in App Store:  3222
Number of English, free apps in Google Play:  8863


As the assignment states our developers are creating apps that are free to download and use. The revenue is created by the in-app adds. The more people are using the app, more people are seeing the add, the bigger the revenue is. 

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:

1. Build a minimal Android version of the app, and add it to Google Play.
2. If the app has a good response from users, we develop it further.
3. If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.

To further analyze we are looking for apps that are successful on both mobile platforms

## Most common app genres

To analyze the most common app genres we will work with column prime_genre in the App Store dataset and with columns Genres and Category in the Google Play Store dataset. 

To know what are the most popular apps we will create frequency tables.

In [21]:
def freq_table(dataset, index):
    frequency_table = {}
    total = 0
    
    for row in dataset:
        total += 1
        data_point = row[index]
        if data_point in frequency_table:
            frequency_table[data_point] += 1
        else:
            frequency_table[data_point] = 1
            
    table_percentages = {}
    for key in frequency_table:
        percentage = (frequency_table[key] / total) * 100
        table_percentages[key] = percentage
    
    return table_percentages

def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

### Frequency table for the App Store
Displaying the frequency talbe for free, english apps in the Apple App Store.

In [22]:
display_table(apple_free, 11) # Display column prime_genre in App Store

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


The most common app genre in the App store is **Games** with more than half of the occurences (58%). Followed by Entertainment 7.8% and Photo & Video 4.9%.

Most of the free English apps are designed for fun (games, entertainment, photo & video, social networking).

### Frequency table for the Google Play
Displaying the frequency talbe for free, english apps in the Google Play Store.

In [23]:
display_table(google_free, 9 ) # Display column Genres in Play Store

Tools : 8.450863138892023
Entertainment : 6.070179397495204
Education : 5.348076272142616
Business : 4.592124562789123
Productivity : 3.8925871601038025
Lifestyle : 3.8925871601038025
Finance : 3.7007785174320205
Medical : 3.5315355974275078
Sports : 3.463838429425702
Personalization : 3.317161232088458
Communication : 3.2381812027530184
Action : 3.102786866749408
Health & Fitness : 3.0802211440821394
Photography : 2.944826808078529
News & Magazines : 2.798149610741284
Social : 2.6627552747376737
Travel & Local : 2.324269434728647
Shopping : 2.245289405393208
Books & Reference : 2.1437436533904997
Simulation : 2.042197901387792
Dating : 1.8616721200496444
Arcade : 1.8503892587160102
Video Players & Editors : 1.771409229380571
Casual : 1.7601263680469368
Maps & Navigation : 1.399074805370642
Food & Drink : 1.241114746699763
Puzzle : 1.128286133363421
Racing : 0.9928917973598104
Role Playing : 0.9364774906916393
Libraries & Demo : 0.9364774906916393
Auto & Vehicles : 0.9251946293580051
S

In [24]:
display_table(google_free, 1) # Display column Category in Play Store

FAMILY : 18.898792733837304
GAME : 9.725826469592688
TOOLS : 8.462146000225657
BUSINESS : 4.592124562789123
LIFESTYLE : 3.9038700214374367
PRODUCTIVITY : 3.8925871601038025
FINANCE : 3.7007785174320205
MEDICAL : 3.5315355974275078
SPORTS : 3.396141261423897
PERSONALIZATION : 3.317161232088458
COMMUNICATION : 3.2381812027530184
HEALTH_AND_FITNESS : 3.0802211440821394
PHOTOGRAPHY : 2.944826808078529
NEWS_AND_MAGAZINES : 2.798149610741284
SOCIAL : 2.6627552747376737
TRAVEL_AND_LOCAL : 2.335552296062281
SHOPPING : 2.245289405393208
BOOKS_AND_REFERENCE : 2.1437436533904997
DATING : 1.8616721200496444
VIDEO_PLAYERS : 1.7939749520478394
MAPS_AND_NAVIGATION : 1.399074805370642
FOOD_AND_DRINK : 1.241114746699763
EDUCATION : 1.1621347173643235
ENTERTAINMENT : 0.9590432133589079
LIBRARIES_AND_DEMO : 0.9364774906916393
AUTO_AND_VEHICLES : 0.9251946293580051
HOUSE_AND_HOME : 0.8236488773552973
WEATHER : 0.8010831546880289
EVENTS : 0.7108202640189552
PARENTING : 0.6544059573507841
ART_AND_DESIGN : 0

Google play store have more apps that are designed for practical purposes. The inner structure of the Play store is different than in the app store. The categories are showing apps in specific catergories like Games or Social. Genres is showing more granula data. We can tell that the category Games is including genres such as Action, Aracade, Puzzle, etc. 

## Finding most popular genres
We now know what are the most common apps. We would like to know what are the most popular apps in terms of downloads. To find those apps we will use column Installs in the google datasets. App Store does not have such column. As a workaround we use the rating_count_column to get approximate values of downloads. Note that not everyone who downloaded and installed the app also rate it, so the actuall number of installs on App store is most likely higher.

### App store

In [25]:
apple_prime_genres = freq_table(apple_free, 11)

In [26]:
for genre in apple_prime_genres:
    total = 0
    len_genre = 0
    for row in apple_free:
        genre_app = row[11]
        if genre_app == genre:
            n_ratings = float(row[5])
            total += n_ratings
            len_genre +=1
    avg_n_rating = total / len_genre
    print(genre, " : ", round(avg_n_rating))
            
    

Social Networking  :  71548
Photo & Video  :  28442
Games  :  22789
Music  :  57327
Reference  :  74942
Health & Fitness  :  23298
Weather  :  52280
Utilities  :  18684
Travel  :  28244
Shopping  :  26920
News  :  21248
Navigation  :  86090
Lifestyle  :  16486
Entertainment  :  14030
Food & Drink  :  33334
Sports  :  23009
Book  :  39758
Finance  :  31468
Education  :  7004
Productivity  :  21028
Business  :  7491
Catalogs  :  4004
Medical  :  612


Let's analyze the data even more and see what apps are under each genre. We will focus on the apps that have a lot of average number of ratins to find long tail values.

In [27]:
def inspect_genre(dataset, string, index_name, index_n_rating, index_genre):
    for row in dataset:
        name = row[index_name]
        n_rating = row[index_n_rating]
        genre = row[index_genre]
        if genre == string:
            print(name, ":", n_rating)

In [28]:
inspect_genre(apple_free, "Social Networking", 1, 5, 11)

Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, Meet : 39819
Hangouts : 36404
LINE PLAY - Your Avatar World : 34677
WeChat : 34584
Badoo - Meet New People, Chat, Socialize. : 34428
Followers + for Instagram - Follower Analytics : 28633
GroupMe : 28260
Marco Polo Video Walkie Talkie : 27662
Miitomo : 2

In [29]:
inspect_genre(apple_free, "Navigation", 1, 5, 11)

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


In [30]:
inspect_genre(apple_free, "Reference", 1, 5, 11)

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


In [31]:
inspect_genre(apple_free, "Photo & Video", 1, 5, 11)

Instagram : 2161558
Snapchat : 323905
YouTube - Watch Videos, Music, and Live Streams : 278166
Pic Collage - Picture Editor & Photo Collage Maker : 123433
Funimate video editor: add cool effects to videos : 123268
musical.ly - your video social network : 105429
Photo Collage Maker & Photo Editor - Live Collage : 93781
Vine Camera : 90355
Google Photos - unlimited photo and video storage : 88742
Flipagram : 79905
Mixgram - Picture Collage Maker - Pic Photo Editor : 54282
Shutterfly: Prints, Photo Books, Cards Made Easy : 51427
Pic Jointer – Photo Collage, Camera Effects Editor : 51330
Color Pop Effects - Photo Editor & Picture Editing : 45320
Photo Grid - photo collage maker & photo editor : 40531
iSwap Faces LITE : 39722
MOLDIV - Photo Editor, Collage & Beauty Camera : 39501
Photo Editor by Aviary : 39501
Photo Lab: Picture Editor, effects & fun face app : 34585
Rookie Cam - Photo Editor & Filter Camera : 33921
FotoRus -Camera & Photo Editor & Pic Collage Maker : 32558
PicsArt Photo St

Even though the games are making up to 58% of the App store, they are not the most popular on the App store. Most popular apps are Social Networking, Navigation and Reference. To our potential developer team I would recommend to start developing a Photo & Video app, that is the second most popular and is third most common. Developing this kind of app can be linked to the enormous popularity of social media apps. On the social media people are sharing photos & videos, so our app can be link in the social-media sharing process. 

### Google Play

The Google Play store has an information about the number of installs. However the number is not precise, rather it is store in buckets.

In [32]:
display_table(google_free, 5)

1,000,000+ : 15.728308699086089
100,000+ : 11.55365000564143
10,000,000+ : 10.549475346947986
10,000+ : 10.199706645605326
1,000+ : 8.394448832223853
100+ : 6.916393997517771
5,000,000+ : 6.826131106848697
500,000+ : 5.562450637481666
50,000+ : 4.772650344127271
5,000+ : 4.513144533453684
10+ : 3.542818458761142
500+ : 3.2494640640866526
50,000,000+ : 2.3017037120613786
100,000,000+ : 2.1324607920568655
50+ : 1.9180864267178157
5+ : 0.7898002933543946
1+ : 0.5077287600135394
500,000,000+ : 0.270788672007221
1,000,000,000+ : 0.2256572266726842
0+ : 0.045131445334536835


For the purpose of this project we will convert the 5+ installs as 5, 50+ as 50, 10,000+ as 10000 and so on. To do that we need to remove the special characters **+**.

In [33]:
google_pop_genre = freq_table(google_free,1)

In [34]:
for category in google_pop_genre:
    total = 0
    len_category = 0
    for row in google_free:
        category_app = row[1]
        if category_app == category:
            n_installs = row[5]
            n_installs = n_installs.replace(",","")
            n_installs = n_installs.replace("+","")
            n_installs = float(n_installs)
            total += n_installs
            len_category +=1
    avg_n_installs = total / len_category
    print(category, ":", round(avg_n_installs))
    

ART_AND_DESIGN : 1986335
AUTO_AND_VEHICLES : 647318
BEAUTY : 513152
BOOKS_AND_REFERENCE : 8767812
BUSINESS : 1712290
COMICS : 817657
COMMUNICATION : 38456119
DATING : 854029
EDUCATION : 1833495
ENTERTAINMENT : 11640706
EVENTS : 253542
FINANCE : 1387692
FOOD_AND_DRINK : 1924898
HEALTH_AND_FITNESS : 4188822
HOUSE_AND_HOME : 1331541
LIBRARIES_AND_DEMO : 638504
LIFESTYLE : 1437816
GAME : 15588016
FAMILY : 3697848
MEDICAL : 120551
SOCIAL : 23253652
SHOPPING : 7036877
PHOTOGRAPHY : 17840110
SPORTS : 3638640
TRAVEL_AND_LOCAL : 13984078
TOOLS : 10801391
PERSONALIZATION : 5201483
PRODUCTIVITY : 16787331
PARENTING : 542604
WEATHER : 5074486
VIDEO_PLAYERS : 24727872
NEWS_AND_MAGAZINES : 9549178
MAPS_AND_NAVIGATION : 4056942


Apps in communication category have 38,456,119 installs. By far those apps are the most installed on Google Play. Games apps have 15,588,016 installs. The category photography 17,840,110 and Social is 23,253,652.
    